lib/doorkeeper/client_authentication.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "doorkeeper/client_authentication/flow"
+require "doorkeeper/client_authentication/registry"
+
+module Doorkeeper
+  module ClientAuthentication
+    extend Registry
+
+    register(
+      :none,
+      mechanism: Doorkeeper::ClientAuthentication::Mechanisms::None,
+      authenticates_client: false
```

Comment:
I'm not sure if I really need this, since we have `blank?` on `Doorkeeper::ClientAuthentication::Credentials`